launcher/agent/agent.go

```diff
@@ -100,27 +120,32 @@ func CreateAttestationAgent(tpm io.ReadWriteCloser, akFetcher util.TpmKeyFetcher
 		sigsCache:        &sigsCache{},
 	}
 
+	// Add TPM
+	logger.Info("Adding TPM PCRs for measurement.")
+	attestAgent.ar = append(attestAgent.ar, &tpmAttestRoot{
+		fetchedAK: ak,
+		tpm:       tpm,
+	})
+
 	// check if is a TDX machine
 	qp, err := tg.GetQuoteProvider()
-	if err != nil || qp.IsSupported() != nil {
-		logger.Println("Using TPM PCRs for measurement.")
+	if err != nil {
+		return nil, err
+	}
+	// Use qp.IsSupported to check the TDX RTMR interface is enabled
+	if qp.IsSupported() == nil {
```

Comment:
Here, we should create a list of roots of trust to measure to (e.g., `var measuredRots []attestRoot`) and then select the RTMR for the actual attestation verification root of trust (e.g., `avRot`).

```
var tpmAR := &tpmAttestRoot{...}
var measuredRots []*attestRoot
var avRot *attestRoot
// Check if VM is running TDX
...
if qp.IsSupported() == nil {
  tdxAR := &tdxAttestRoot
  measuredRots = append(measuredRots, tdxAR)
  avRot = tdxAr
} else {
  avRot = tpmAR
}
// Unconditionally add TPM.
measuredRots = append(measuredRots, tpmAR)
```

Reply:
Done